src/main/clojure/xtdb/test_generators.clj

```diff
@@ -121,3 +133,92 @@
              field-values (gen/vector recursive-value-gen num-fields)]
      (-> (zipmap field-keys field-values)
          (assoc :xt/id id)))))
+
+(def single-type-vector-vs-gen 
+  (fn [min-length max-length]
+    (gen/let [vector-name (gen/such-that #(not (str/blank? %)) gen/string-alphanumeric 100)
+              type-gen (gen/elements simple-type-gens)
+              vs (gen/vector type-gen min-length max-length)]
+      {:vec-name vector-name
+       :type type-gen
+       :vs vs})))
+
+(def dense-vector-vs-gen
+  (fn [min-length max-length]
+    (gen/let [vector-name (gen/such-that #(not (str/blank? %)) gen/string-alphanumeric 100)
+              vs (gen/vector recursive-value-gen min-length max-length)]
+      {:vec-name vector-name 
+       :vs vs})))
+
+(def vector-vs-gen
+  (gen/frequency
+   [[3 (single-type-vector-vs-gen 1 100)]
+    [1 (dense-vector-vs-gen 1 100)]]))
+
+(def fixed-length-vector-vs-gen 
+  (fn [length]
+    (gen/frequency
+     [[3 (single-type-vector-vs-gen length length)]
+      [1 (dense-vector-vs-gen length length)]])))
+
+(def two-distinct-single-type-vecs-gen
+  (gen/let [vec1 (single-type-vector-vs-gen 1 100)
+            vec2 (gen/such-that #(and (not= (:type %) (:type vec1))
+                                      (not= (:vec-name %) (:vec-name vec1)))
+                                (single-type-vector-vs-gen 1 100)
+                                100)]
+    [vec1 vec2]))
+
+(def two-distinct-duvs-gen
+  (gen/let [duv1 (dense-vector-vs-gen 1 100)
+            duv2 (gen/such-that #(not= (:vec-name %) (:vec-name duv1))
+                                (dense-vector-vs-gen 1 100)
+                                100)]
+    [duv1 duv2]))
+
+(defn vec-gen->arrow-vec [^BufferAllocator allocator {:keys [^String vec-name ^List vs]}]
+  (Vector/fromList allocator vec-name vs))
+
+;; TODO - Any utils/other way we could handle this?
+(defn normalize-for-comparison
+  [obj]
+  (cond
```

Comment:
Nit: Seems like a weird function indentation. I only use that if it is a varargs function.